### PR TITLE
Fix dump scenario command and add tests

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR fixes the dump scenario query, which was wrong because it was considering only the latest event from the projector subscriptions, while a projector can handle multiple events, resulting in an incomplete scenario dump.

It also adds tests.

### CODE REVIEW

File: cmd/ctl/ctl_test.go [hunk 1]
```diff
@@ -103,3 +103,51 @@ func (suite *CtlTestSuite) TestPruneChecksResults() {
...
+			AgentID:       "agent_id_2",
+			DiscoveryType: "discovery_type_2",
+			Payload:       []byte("{}"),
+		},
+		{
+			ID:            5,
+			AgentID:       "agent_id_2",
+			DiscoveryType: "discovery_type_2",
+			Payload:       []byte("{}"),
+		},
```
Comment (arbulu89) [inline cmd/ctl/ctl_test.go:141]: Can we add an additional discovery here in the test with a different discovery type?
To test if the latest event with other discovery type is retrieved as well.
Simply add a new entry with `discovery_type_3` for any of this 2 agents, and check if it is obtained

**Review by arbulu89 — APPROVED** [review]
Thanks!
To understand it better.
The 1st subquery gets all the ids for each `agent_id/discovery_type` tuple, right?
PD: I suggest to improve a bit the added test to include a normal use case, which is to have other discovery types per agent

**Review by dottorblaster — APPROVED** [review]
LGTM

**Review by nelsonkopliku — APPROVED** [review]

**Comment by fabriziosestito** [comment]
> Thanks! To understand it better. The 1st subquery gets all the ids for each `agent_id/discovery_type` tuple, right? PD: I suggest to improve a bit the added test to include a normal use case, which is to have other discovery types per agent

The subquery select the max id from after grouping by agent_id and discovery_type. The max is important since it is the latest one.
The other query just select everything that has the id returned by the subquery.

File: cmd/ctl/ctl_test.go (hunk 1 above)
Comment (fabriziosestito) [reply cmd/ctl/ctl_test.go:141]: @arbulu89 test added

**Pull request merged**